velox/substrait/tests/SubstraitVeloxPlanConverterTest.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <folly/Random.h>
+
+#include "velox/exec/tests/utils/OperatorTestBase.h"
+#include "velox/exec/tests/utils/PlanBuilder.h"
+#include "velox/vector/tests/VectorMaker.h"
+
+#include "velox/substrait/SubstraitToVeloxPlan.h"
+#include "velox/substrait/VeloxToSubstraitPlan.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+using namespace facebook::velox::exec;
+using namespace facebook::velox::exec::test;
+using namespace facebook::velox::substrait;
+
+class SubstraitVeloxPlanConvertorTest : public OperatorTestBase {
+ protected:
+  /// the function used to make data
+  /// \param size the number of RowVectorPtr.
+  /// \param childSize the size of rowType_, the number of columns
+  /// \param batchSize batch size.  row number
+  /// \return std::vector<RowVectorPtr>
+  std::vector<RowVectorPtr>
+  makeVector(int64_t size, int64_t childSize, int64_t batchSize) {
+    std::vector<RowVectorPtr> vectors;
+    std::mt19937 gen(std::mt19937::default_seed);
+    for (int i = 0; i < size; i++) {
+      std::vector<VectorPtr> children;
+      for (int j = 0; j < childSize; j++) {
+        children.emplace_back(makeFlatVector<int32_t>(
+            batchSize,
+            [&](auto row) {
+              return folly::Random::rand32(INT32_MAX / 4, INT32_MAX / 2, gen);
+            },
+            nullEvery(2)));
+      }
+
+      std::vector<std::string> childrenNames = rowType_->names();
+      vectors.push_back(makeRowVector({childrenNames}, {children}));
+    }
+    return vectors;
+  };
+
+  void assertPlanConversion(
+      const std::shared_ptr<const core::PlanNode>& vPlan,
+      const std::string& duckDbSql) {
+    assertQuery(vPlan, duckDbSql);
+    // Convert Velox Plan to Substrait Plan
+    auto arena = std::make_shared<google::protobuf::Arena>();
+    v2SPlanConvertor_.toSubstrait(*arena, vPlan);
```

Comment:
Sure. But to make this PR not be more complex, I create PR https://github.com/facebookincubator/velox/pull/1437  in another thread to add the convertor from Substrait VirtualTable to Velox ValuesNode. So that the round-trip test will be blocked by the PR  https://github.com/facebookincubator/velox/pull/1437, could you please help review  https://github.com/facebookincubator/velox/pull/1437 ? or we just add them into this PR?  I have tested the code locally, it can pass the round-trip.